envi/tests/test_arch_arm.py

```diff
@@ -5,68 +5,593 @@
+        (REV_ALL_ARM, '3746d3e0', 0x4560, 'sbcs r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746e3e0', 0x4560, 'rsc r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746f3e0', 0x4560, 'rscs r4, r3, r7, lsr r6', 0, ()),
+        #(REV_ALL_ARM, '374603e1', 0x4560, 'tstS r3, r7, lsr r6', 0, ()),   #added s   # not implimented
+        (REV_ALL_ARM, '374613e1', 0x4560, 'tsts r3, r7, lsr r6', 0, ()),  #added s
+        (REV_ALL_ARM, '374623e1', 0x4560, 'blx r7', 0, ()),
+        (REV_ALL_ARM, '374633e1', 0x4560, 'teqs r3, r7, lsr r6', 0, ()), #added s
+        #(REV_ALL_ARM, '374643e1', 0x4560, 'cmps r3, r7, lsr r6', 0, ()),    #added s   # not implimented
+        (REV_ALL_ARM, '374653e1', 0x4560, 'cmps r3, r7, lsr r6', 0, ()), #added s
+        #(REV_ALL_ARM, '374663e1', 0x4560, 'cmns r3, r7, lsr r6', 0, ()),   #added s   # not implimented
+        (REV_ALL_ARM, '374673e1', 0x4560, 'cmns r3, r7, lsr r6', 0, ()),  #added s 
+        (REV_ALL_ARM, '374683e1', 0x4560, 'orr r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '374693e1', 0x4560, 'orrs r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746a3e1', 0x4560, 'mov r4, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746b3e1', 0x4560, 'movs r4, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746c3e1', 0x4560, 'bic r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746d3e1', 0x4560, 'bics r4, r3, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746e3e1', 0x4560, 'mvn r4, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '3746f3e1', 0x4560, 'mvns r4, r7, lsr r6', 0, ()),
+        (REV_ALL_ARM, '474603e0', 0x4560, 'and r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474613e0', 0x4560, 'ands r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474623e0', 0x4560, 'eor r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474633e0', 0x4560, 'eors r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474643e0', 0x4560, 'sub r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474653e0', 0x4560, 'subs r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474663e0', 0x4560, 'rsb r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474673e0', 0x4560, 'rsbs r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474683e0', 0x4560, 'add r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '474693e0', 0x4560, 'adds r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746a3e0', 0x4560, 'adc r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746b3e0', 0x4560, 'adcs r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746c3e0', 0x4560, 'sbc r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746d3e0', 0x4560, 'sbcs r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746e3e0', 0x4560, 'rsc r4, r3, r7, asr #12', 0, ()),
+        (REV_ALL_ARM, '4746f3e0', 0x4560, 'rscs r4, r3, r7, asr #12', 0, ()),
+        #(REV_ALL_ARM, '474603e1', 0x4560, 'tsts r3, r7, asr #12', 0, ()),  #added s   # should be: tsts r3, r7, asr #12  - is: mrs r4, CPSR
```

Comment:
add new flag:  IF_SILENT_S
in combination with IF_S (which will also be set).  the IF_S has to stay, because it's used by analysis for more than just repr()/render().  for cmp, teq, tst or anything else in the future that leaves off the "s" because it's assumed, we'll have the Opcode.repr/render functions watch for IF_SILENT_S and simply not add it.

i haven't looked too heavily into UAL, but it's possible that some of the reasons for our oddball repr/render approachs for mnems is going away.  we'll have to review this when done implementing/fixing all the stuff.